content/guides/end-to-end-testing/testing-your-app.md

```diff
@@ -410,111 +410,87 @@ You'll likely also want to test your login UI for:
-browser!
+You're probably wondering what happened to our advice about logging in "only
+once". The custom command above will work just fine for testing your secured
+pages, but if you have more than a handful of tests, logging in before every
+test is going to add a lot of running time to your suite.
 
-Let's revisit the example from above but assume we're testing some other part of
-the system.
+Luckily, Cypress provides the [`cy.session()` command](/api/commands/session), a
+powerful performance tool that lets you cache user session data and reuse it for
+multiple tests without going through multiple login flows! Let's modify the
+`cy.login()` command from our previous example to use `cy.session()`:
 
 ```js
-describe('The Dashboard Page', () => {
-  beforeEach(() => {
-    // reset and seed the database prior to every test
-    cy.exec('npm run db:reset && npm run db:seed')
-
-    // seed a user in the DB that we can control from our tests
-    // assuming it generates a random password for us
-    cy.request('POST', '/test/seed/user', { username: 'jane.lane' })
-      .its('body')
-      .as('currentUser')
-  })
-
-  it('logs in programmatically without using the UI', function () {
-    // destructuring assignment of the this.currentUser object
-    const { username, password } = this.currentUser
-
-    // programmatically log us in without needing the UI
-    cy.request('POST', '/login', {
-      username,
-      password,
-    })
-
-    // now that we're logged in, we can visit
-    // any kind of restricted route!
-    cy.visit('/dashboard')
-
-    // our auth cookie should be present
-    cy.getCookie('your-session-cookie').should('exist')
-
-    // UI should reflect this user being logged in
-    cy.get('h1').should('contain', 'jane.lane')
-  })
+Cypress.Commands.add('login', (username, password) => {
+  cy.session(
+    username,
+    () => {
+      cy.visit('/login')
+      cy.get('input[name=username]').type(username)
+      cy.get('input[name=password]').type(`${password}{enter}`)
+      cy.url().should('include', '/dashboard')
+      cy.get('h1').should('contain', username)
+    },
+    {
+      validate: () => {
+        cy.getCookie('your-session-cookie').should('exist')
+      },
+    }
+  )
 })
 ```
 
-Do you see the difference? We were able to login without needing to actually use
-our UI. This saves an enormous amount of time visiting the login page, filling
-out the username, password, and waiting for the server to redirect us _before
-every test_.
-
-Because we previously tested the login system end-to-end without using any
-shortcuts, we already have 100% confidence it's working correctly.
-
-Use the methodology above when working with any area of your system that
-requires the state to be set up elsewhere. Just remember - don't use your UI!
+There's a lot going on here that's out of the scope of this introduction, so for
+a more in-depth explanation of how `cy.session()` works,
+[read the API docs](/api/commands/session).
```

Comment:
```suggestion
[read the commands docs](/api/commands/session).
```
Cypress API is different than Cypress commands since its sync helpers that execute immediately instead of being queue to run later like Cypress commands

Reply:
I usually refer to everything underneath the "API" top-level docs tab as "the API docs". I know that there is also (confusingly) something called the "Cypress API" which is different, but I'm not referring to that. I do see your point but I think it needs to be consistent, if we're going to change this it needs to be "command docs" everywhere